### DISCUSSION TITLE
O futuro do front-end é design

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#382] [Discussão]
Eu sempre pensei dessa forma.. mas de uns 4,5 anos pra cá parece que é o contrario e a área se separou em nichos.. e pelo que vejo em questão de vagas os front-end engineers são os que ganham mais.

Mas esse artigo do Tech Crunch é bem interessante:
https://techcrunch.com/2016/11/29/the-future-of-front-end-development-is-design/

o que acham?

Eu no momento to bem focado em UI/UX e pretendo também me profissionalizar mais em criação e etc que eu ja tenho uma noção e um portfolio muito maior do que coisas que envolvam back e javascript puro ( que é algo que eu preciso me aprofundar tambem )

Mas como vocês vem a área daqui, 2 ,3, 5 anos?

**Comment by rubenmarcus** [comment]
Uma percepção bem minha que sou mais ligado ao design, é que quando vejo um portfolio de um front-end com template pronto, parece um pouco relaxo, e tambem descaso com design.. Mas tambem não posso argumentar quando vejo programadores que tem vários scripts, participação em open source, github cheio, e não tem nem portfolio..

Mas pelo menos pra mim.. me miro muito nisso:

http://claudiocalautti.cc/

Penso que cada programador é unico e tem uma forma de fazer o proprio código,nomear classes e etc.. Também deveria ser assim na parte criativa, que não é pouca na nossa profissão

**Comment by woliveiras** [comment]
Não me vejo desenhando nada.

Meu prazer em Desenvolvimento Front End é desenvolver softwares que rodam no client side. Se me der uma tela em branco e um briefing de um cliente eu, com certeza, faço algo feio! :joy: :joy: :joy:

Talvez uma separação em dois tipos de Front End, um que cuida de Design e outro que cuida de Programação, como já foi discutido diversas vezes (#32 tem algumas referências) nesse forum e rendeu até um repositório: https://github.com/woliveiras/front-end-career, seja o ideal.

Dizer que o FUTURO de Front End é Design por que nasceram ferramentas ou por que algumas pessoas estão indo para essa área ou dizer que o FUTURO de Front End é programação por que nasceram outras ferramentas e outras pessoas tendem a essa área eu acho errado.

Penso eu: se é uma área que da pra ser dividida, então que dividamos e cada um vá fazer o que realmente curte, não que digamos que o futuro de uma profissão é só o que eu gosto.

Se for assim, como no título, fica parecendo que eu ou pessoas com um perfil parecido com o meu deveriam abandonar Front End e virar Back End ou Androidero/iOSzero que é tipo Front, mas gourmetizado (e ganha mais do que o tal Front End Engineer).

Eu curto escrever CSS e me preocupar com GPU, performance, curto escrever HTML e me preocupar com acessibilidade, SEO (por mais que eu não curta muito esse ponto), curto escrever JS e me preocupar com arquitetura, escalabilidade. Gosto de misturar tudo isso e pensar em uma tela, pensar em componentes, escalabilidade, mas não gostaria que alguém me dissesse que eu sou Designer (ou continuasse a dizer, pois muita gente me pede pra desenhar temas/layouts) por eu mexer com Front, afinal estaria me confundindo com quem trabalha com Design e tem preocupações diferentes das minhas.

Eu sei sobre tipografia, um pouco de teoria das cores, experiência de usuário, porém minha experiência não chega nem aos pés do conhecimento do Designer que cria as telas que eu codifico no meu emprego.

**Comment by bernardodiasc** [comment]
Vale ressaltar que no artigo o Tech Crunch, quando o autor diz "design" está se referindo à "product design". Não é exatamente a função de criar telas no photoshop, tem mais a ver com o _projetar o produto_ (o significado de __design__ é __projeto__ e não é necessariamente relacionado à direção de arte).

O apelo do titulo remete ao fato de que com tantas ferramentas de construção de sites, hoje em dia existem possibilidades de se lançar produtos digitais sem precisar ter especialização em código, de modo que um "product designer" pode criar e produzir suas próprias ideias.

**Comment by rubenmarcus** [comment]
@woliveiras eu concordo com você! O fato de eu pensar na parte visual, e também na parte de design (projeto) é que eu sempre gostei disso bastante. 

Em relação ao mercado, vocês acham que, os profissionais num geral serão mais valorizados, essas ferramentas que o artigo cita, vai ampliar o numero de profissionais, ou diminui-los? O mercado vai crescer ainda mais?

**Comment by danilowoz** [comment]
Como designer de formação e front-end autodidata que me tornei, acredito que a essência do design esta intrínseco no nosso dia-a-dia, ao pensar na acessibilidade, usabilidade, experiência do usuário (que pode ser uma página carregando mais rápida, ou prever ações) e principalmente em resolver problemas, que acredito que seja o verdadeiro significado de design.

E hoje em dia as ferramentas estão se tornando cada vez mais práticas, fáceis de usar e aplicar... E vem conquistando um novo tipo de profissional e atendendo outro público (devs) que já sentia a necessidade de criar atalhos, ferramentas para aplicações visuais e experiências mais fluídas para os usuários.

Assim como o autor coloca, acredito que estamos construindo caminhos para aproximar as áreas, que sempre houve um histórico de "conflito" (põe um 1px pra esquerda!, não ta bold!, melhora o kerning!) e criar ainda mais especialistas.

**Comment by vitocaio** [comment]
Acho muito interessante falar novamente sobre esse assunto por esse ponto de vista, hoje a preocupação do Front não é mais só o Client-Side de modo genérico, você tem vários seguimentos dentro de seguimentos. Gerando demandas de profissionais especialista nestes micro seguimentos, performance, escalabilidade, usabilidade etc...
O que quero dizer é, acho que nossa amada carreia de Front, nunca foi tão bem definida como esta sendo agora, é muito interessante pensar que, daqui alguns anos ou até antes, as empresas irão ter equipes de Front com 3 ou 4 integrantes, todos pensando somente no Client-side e não mais um cara só que manja tudo e as vezes até de Back, o famoso "the choose one", kkk .

**Comment by fernandoRj** [comment]
o que eu vejo hoje em dia e as empresas contratando front end elas pedem html css javascript jquery e linguagens como angular, react, manjar de api

e outras vagas voltadas para web designer que a grande maioria exige photoshop, e outras ferramentas de designer + linguanges html css javascript e jquery 

ou seja hoje em dia no mercado existem 2 tipos de front, um que alem de criar a interface cria aplicaçoes (esse nao precisa manjar mt de designer) com angular ou outra linguagem

 e o front end mais voltado para designer que cria interfaces lindas
